frontend/tests/controllers-test/mainCtrl.test.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+describe('Unit tests for Main Controller', function () {
+    beforeEach(angular.mock.module('evalai'));
+
+    var $controller, createController, $rootScope, $scope, utilities, $state, vm;
+
+    beforeEach(inject(function (_$controller_, _$rootScope_, _$state_, _utilities_,) {
+        $controller = _$controller_;
+        $rootScope = _$rootScope_;
+        utilities = _utilities_;
+        $state = _$state_;
+
+        $scope = $rootScope.$new();
+        createController = function () {
+            return $controller('MainCtrl', {$scope: $scope});
+        };
+        vm = createController();
+    }));
+
+    describe('Global variables', function () {
+        it('has default values', function () {
+            vm = createController();
+            expect(vm.user).toEqual({});
+            expect(vm.challengeList).toEqual([]);
+            expect(vm.isChallenge).toBeTruthy({});
+            expect(vm.isMore).toBeFalsy();
+            expect(vm.redirectUrl).toEqual({});
+        });
+    });
+
+    describe('Unit tests for getChallenge function `challenges/featured/`', function () {
+        var success, successResponse;
+        var errorResponse = 'error';
+
+        beforeEach(function () {
+            utilities.sendRequest = function (parameters) {
+                if (success) {
+                    parameters.callback.onSuccess({
+                        data: successResponse
+                    });
+                } else {
+                    parameters.callback.onError({
+                        data: errorResponse
+                    });
+                }
+            };
+        });
+
+        it('successfully get featured challenge', function () {
+            success = true;
+            successResponse = 'success';
+            vm.getChallenge();
+            expect(vm.challengeList).toEqual(successResponse);
+        });
+    });
+
+    describe('Unit tests for init function `auth/user/`', function () {
+        var success, successResponse, errorResponse;
+
+        beforeEach(function () {
+            spyOn($state, 'go');
+            spyOn(utilities, 'resetStorage');
+            utilities.storeData('userKey', 'encrypted key');
+
+            utilities.sendRequest = function (parameters) {
+                if (success) {
+                    parameters.callback.onSuccess({
+                        data: successResponse,
+                        status: 200
+                    });
+                } else {
+                    parameters.callback.onError({
+                        data: errorResponse,
+                        status: 401
+                    });
+                }
+            };
+        });
+
+        it('check for the authenticated user', function () {
+            success = true;
+            successResponse = {
+                username: 'abc123'
+            };
+            vm.init();
+            expect(vm.user.name).toEqual(successResponse.username);
```

Comment:
on success you can validate resetStorage is not being called.

Reply:
done